### PR TITLE
basic indexing for models and metastreams

### PR DESCRIPTION
**Pull request by ebenenglish**
* adds basic Solr indexing functionality for:
  * `Curator::Collection`
  * `Curator::DigitalObject`
  * `Curator::Filestreams::FileSet`
  * `Curator::Metastreams::Workflow `
  * `Curator::Metastreams::Administrative`

* adds some convenience methods to model classes to make indexing of associations easier

* specs for everything

* `traject` is updated to `3.3.0` to support indexing via included modules

(_Note:_ There are still a lot of TODOs for indexing (most notably `Curator::Metastreams::Descriptive`), going to submit PRs for this in stages to make review easier.)

### CODE REVIEW

File: app/models/curator/digital_object.rb [hunk 1]
```diff
@@ -33,6 +34,17 @@ class DigitalObject < ApplicationRecord
       has_one :issue_for, through: :issue_mapping_for, source: :digital_object
     end
 
+    self.curator_indexable_mapper = Curator::DigitalObjectIndexer.new
+
+    def institution
+      admin_set.institution
+    end
+
```
Comment (bbarberBPL) [inline app/models/curator/digital_object.rb]: replace institution method to

`has_one :institution, through: :admin_set, class_name: 'Curator::Institution`  Move this Just under the `belongs_to admin_set` declaration around L12

File: app/models/curator/collection.rb [hunk 2]
```diff
@@ -6,11 +6,19 @@ class Collection < ApplicationRecord
...
     has_many :admin_set_objects, inverse_of: :admin_set, class_name: 'Curator::DigitalObject', foreign_key: :admin_set_id, dependent: :destroy
 
     has_many :collection_members, inverse_of: :collection, class_name: 'Curator::Mappings::CollectionMember', dependent: :destroy
+
+    self.curator_indexable_mapper = Curator::CollectionIndexer.new
+
+    def exemplary_file_set
+      exemplary_image_mapping = exemplary_image_mappings.first
+      exemplary_image_mapping ? exemplary_image_mapping.exemplary_file_set : nil
+    end
```
Comment (bbarberBPL) [inline app/models/curator/collection.rb]: Move `emplary_file_set` into `Curator::Mappings::Exemplary::ObjectImagable` concern for now pending #23

File: app/models/curator/digital_object.rb (hunk 1 above)
Comment (ebenenglish) [reply app/models/curator/digital_object.rb]: Fixed via 2075da6378d0901b1da558acbcc128391e49ea15

File: app/models/curator/collection.rb (hunk 2 above)
Comment (ebenenglish) [reply app/models/curator/collection.rb]: Fixed via 2075da6378d0901b1da558acbcc128391e49ea15

(I believe this also fixes #23)

**Pull request merged**